src/modelinstance.cpp

```diff
@@ -237,6 +237,34 @@ Status ModelInstance::loadOVCNNNetwork() {
     return StatusCode::OK;
 }
 
+Status ModelInstance::loadOVCNNNetworkUsingCustomLoader() {
+    spdlog::debug("Try reading model using a custom loader");
+    try {
+        char* xmlBuffer = NULL;
+        int xmlLen = 0;
+        char* binBuffer = NULL;
+        int binLen = 0;
+
+        spdlog::info("loading CNNNetwork for model:{} basepath:{} <> {} version:{}", getName(), getPath(), this->config.getBasePath().c_str(), getVersion());
+        int res = customLoaderInterfacePtr->loadModel(this->config.getName().c_str(),
+            this->config.getBasePath().c_str(),
+            getVersion(),
+            this->config.getCustomLoaderOptionsConfigStr().c_str(),
+            &xmlBuffer, &xmlLen, &binBuffer, &binLen);
+        if (res != 1) {
+            return StatusCode::INTERNAL_ERROR;
+        }
+        std::string strModel(xmlBuffer);
+        std::vector<uint8_t> weights(&binBuffer[0], &binBuffer[binLen]);
+        network = std::make_unique<InferenceEngine::CNNNetwork>(engine->ReadNetwork(strModel,
```

Comment:
Are those U8 etc. hardcoded constant for all models loaded by this plugin? Seems so - if confirmed, please add comment with description.

Reply:
ReadNetwork API with wights as a blob needs the weights to be U8 blob.